app/src/main/java/fi/helsinki/opintoni/service/FeedbackService.java

```diff
@@ -68,4 +123,65 @@ public void insertFeedback(InsertFeedbackRequest request) throws Exception {
         feedback.comment = feedbackDto.comment;
         return getAllFeedback();
     }
+
+    private String createTimestamp() {
+        ZonedDateTime helsinkiNow = ZonedDateTime.now(HELSINKI_ZONE_ID);
+        return helsinkiNow.format(TIMESTAMP_FORMATTER);
+    }
+
+    private String getState(InsertFeedbackRequest request) {
+        return request.metadata.findValue("state").textValue();
+    }
+
+    private String getFaculty(InsertFeedbackRequest request) {
+        JsonNode facultyNode = request.metadata.findValue("faculty");
+        return facultyNode == null ? null : facultyNode.textValue();
+    }
+
+    private String getUserAgent(InsertFeedbackRequest request) {
+        return request.metadata.findValue("userAgent").textValue();
+    }
+
+    private Locale getLocale(InsertFeedbackRequest request) {
+        JsonNode langNode = request.metadata.findValue("lang");
+        String langCode = langNode == null ? "fi" : langNode.textValue();
```

Comment:
"fi" kuuluu jonnekin `static final`iin. Lisäksi null-checkin ja default-arvon saa siististi myös näin `Optional.ofNullable(langNode).map(JsonNode::textValue).orElse(DEFAULT_LANG)`